src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -629,7 +639,12 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (this._standardKernels) {
 			let standardKernels = find(this._standardKernels, kernel => this._defaultKernel && kernel.displayName === this._defaultKernel.display_name);
 			let connectionProviderIds = standardKernels ? standardKernels.connectionProviderIds : undefined;
-			return profile && connectionProviderIds && find(connectionProviderIds, provider => provider === profile.providerName) !== undefined;
+			let providerFeatures = this._capabilitiesService.getCapabilities(profile.providerName);
+			if (connectionProviderIds.length > 0) {
+				this._currentKernelAlias = providerFeatures?.connection.notebookKernelAlias;
```

Comment:
looks like we are setting state (_currentKernelAlias) when checking for a validConnect, is there a better location so set this? (like when kernel changes, perhaps).

Reply:
Chatted offline with Vasu, the main idea is to populate the correct value of _currenKernel and _currentKernelAlias as early in the call stack as possible for his scenario and in a method that is clearly earmarked for this purpose instead of a side effect of another status check call like isValidConnection is. We identified startSession is a suitable location to house a call to new method, just before making isValidConnection call to set these state variables.